lib/AST/Type.cpp

```diff
@@ -5084,19 +5084,10 @@ AnyFunctionType *AnyFunctionType::getTransposeOriginalFunctionType(
     transposeResult = methodType->getResult();
   }
 
-  Type originalResult;
-  if (isCurried) {
-    // If it's curried, then the first parameter in the curried type, which is
-    // the 'Self' type, is the original result (no matter if we are
-    // transposing wrt self or not).
-    originalResult = getParams().front().getPlainType();
-  } else {
-    // If it's not curried, the last parameter, the tangent, is always the
-    // original result type as we require the last parameter of the transposing
-    // function to be the original result.
-    originalResult = transposeParams.back().getPlainType();
-  }
-  assert(originalResult);
+  // The last parameter, the tangent, is always the
```

Comment:
Please take care to fill comment lines up to 100 columns before wrapping:
```suggestion
  // The last parameter, the tangent, is always the original result type as we require the last
```
[`git clang-format`](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) isn't able to fix this (with default settings). I'll fix this now.